grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -112,25 +107,14 @@ private Object executeGraql(Request request, Response response) {
         String keyspace = mandatoryQueryParameter(request, KEYSPACE);
         boolean infer = parseBoolean(mandatoryQueryParameter(request, INFER));
         boolean materialise = parseBoolean(mandatoryQueryParameter(request, MATERIALISE));
+        int limitEmbedded = queryParameter(request, LIMIT_EMBEDDED).map(Integer::parseInt).orElse(-1);
         String acceptType = getAcceptType(request);
+
         try(GraknGraph graph = factory.getGraph(keyspace, WRITE)){
```

Comment:
Worth noting that making this a `READ` transaction when you are indeed only reading has some tiny performance benefits. Basically it allows some more aggressive central caching to be done.